exchanges/kucoin/kucoin_websocket.go

```diff
@@ -1686,13 +1683,41 @@ func channelName(s *subscription.Subscription, a asset.Item) string {
 	return s.Channel
 }
 
-// removeSpotFromMargin removes spot pairs from margin pairs in the supplied AssetPairs map for subscriptions to non-margin endpoints
-func removeSpotFromMargin(s *subscription.Subscription, ap map[asset.Item]currency.Pairs, spotPairs currency.Pairs) string {
+// mergeMarginPairs merges margin pairs into spot pairs for shared subs (ticker, orderbook, etc) if Spot asset and sub are enabled,
+// because Kucoin errors on duplicate pairs in separate subs, and doesn't have separate subs for spot and margin
+func (ku *Kucoin) mergeMarginPairs(s *subscription.Subscription, ap map[asset.Item]currency.Pairs) string {
```

Comment:
```
[ERROR] | WEBSOCKET | 31/12/2024 09:57:40 | Kucoin websocket: subscription failure, ticker all : subscription template did not generate the expected number of asset records: Got 2; Expected 1, orderbook all : subscription template did not generate the expected number of asset records: Got 2; Expected 1
```
I hadn't even tried to mess with things yet. I accidentally had margin asset pair disabled 😭 

Config gist: [here](https://gist.github.com/gloriousCode/8b80a088d462213b40f3b68ddba4a694)
Have I done something wrong?
edit: works when I enable margin...

Reply:
Thanks so much for catching that 🙏  Added a test case.
Need to ensure we don't add an empty margin asset which didn't exist.
Kucoin remains the place we're fangling assetPairs the most, so this is quite edge-casey.

Fixed https://github.com/gbjk/gocryptotrader/commit/a7e1f47fa

Reply:
I've seen another bug now when margin is all that's enabled. I'm on it.

Reply:
Got it 🎉  Just need to leave assets alone in `asset.All` subs unless there's *both* margin and spot subs available.

Fixed https://github.com/gbjk/gocryptotrader/commit/d4ac07b70